lib/sprinkler_web/live/metrics_component.ex

```diff
@@ -9,22 +9,29 @@ defmodule SprinklerWeb.MetricsComponent do
       <div class="metrics-container">
         <div class="metrics-container__metric">
           <h4>Temperature</h4>
-          <div><%= List.first(@device[:tmps]) || "- " %>º</div>
+          <div><%= last_value(@device, :tmps) || "- " %>º</div>
         </div>
         <div class="metrics-container__metric">
           <h4>Humidity</h4>
-          <div><%= List.first(@device[:hum]) || "-" %></div>
+          <div><%= last_value(@device, :hum) || "-" %></div>
         </div>
         <div class="metrics-container__metric">
           <h4>Soil Moisture</h4>
-          <div><%= List.first(@device[:moist]) || "-" %></div>
+          <div><%= last_value(@device, :moist) || "-" %></div>
         </div>
         <div class="metrics-container__metric">
           <h4>Today Irrigations</h4>
-          <div><%= List.first(@device[:irrigations]) || "-" %></div>
+          <div><%= last_value(@device, :irrigations) || "-" %></div>
         </div>
       </div>
     </div>
     """
   end
+
+  defp last_value(metrics, metric_name) do
```

Comment:
😬  I don't like defining functions here. Could we put this somewhere else?  Like a "view helper"

Reply:
If you think about this in the same way as React components, this is a great place to put functions, very close to the code. That being said, this function is definitely much more general than the module. I have some ideas on how to refactor this but I'd rather do it well rather than early

Reply:
This is the way I want to try to refactor this https://hostiledeveloper.com/2020/06/19/organizing-liveview-logic-with-presentation-models.html